src/components/KRadio/KRadio.vue

```diff
@@ -11,105 +12,219 @@
-      default: '',
-    },
-    /**
-     * The value emitted from the radio on change if selected
-     */
-    selectedValue: {
-      type: [String, Number, Boolean, Object],
-      required: true,
+}
+</script>
+
+<script setup lang="ts">
+const props = defineProps({
+  /**
+   * Sets whether or not radio is selected
+   */
+  modelValue: {
+    type: [String, Number, Boolean, Object],
+    default: 'on',
+    required: true,
+  },
+  /**
+   * Overrides default label text
+   */
+  label: {
+    type: String,
+    default: '',
+  },
+  /**
+   * Overrides default description text
+   */
+  description: {
+    type: String,
+    default: '',
+  },
+  /**
+   * The value emitted from the radio on change if selected
+   */
+  selectedValue: {
+    type: [String, Number, Boolean, Object],
+    required: true,
+  },
+  /**
+   * Controls appearance of radio input element
+   */
+  type: {
+    type: String,
+    default: '',
+    validator: (value: string): boolean => {
+      return ['', 'card'].includes(value)
     },
   },
-  emits: ['change', 'update:modelValue'],
-  setup(props, { slots, emit, attrs }) {
-    const hasLabel = computed((): boolean => !!(props.label || slots.default))
+})
 
-    const isSelected = computed((): boolean => props.selectedValue === props.modelValue)
+const slots = useSlots()
 
-    const handleClick = (): void => {
-      emit('change', props.selectedValue)
-      emit('update:modelValue', props.selectedValue)
-    }
+const hasLabel = computed((): boolean => !!(props.label || slots.default))
+const hasDescription = computed((): boolean => !!(props.description || slots.description))
 
-    const modifiedAttrs = computed(() => {
-      const $attrs = { ...attrs }
+const isSelected = computed((): boolean => props.selectedValue === props.modelValue)
 
-      // delete classes because we bind them to the parent
-      delete $attrs.class
+const isTypeDefault = computed((): boolean => props.type === '')
 
-      return $attrs
-    })
+const emit = defineEmits(['change', 'update:modelValue'])
 
-    return {
-      hasLabel,
-      isSelected,
-      modifiedAttrs,
-      handleClick,
-    }
-  },
+const handleClick = (): void => {
+  emit('change', props.selectedValue)
+  emit('update:modelValue', props.selectedValue)
+}
+
+const attrs = useAttrs()
+
+const modifiedAttrs = computed(() => {
+  const $attrs = { ...attrs }
+
+  // delete classes because we bind them to the parent
+  delete $attrs.class
+
+  return $attrs
 })
 </script>
 
-<style lang="scss" scoped>
+<style lang="scss">
 @import '@/styles/variables';
 @import '@/styles/functions';
 
-.k-radio-label {
-  font-size: var(--type-sm, type(sm));
+$text-color-default: var(--black-45, rgba(0, 0, 0, 0.45));
+$text-color-card: color(black-500);
+$border-color-card: color(grey-300);
+$background-color-card-checked: color(blue-100);
+$border-color-card-checked: color(blue-300);
+$background-color-card-disabled: color(grey-200);
+
+@mixin disabled {
```

Comment:
suggestion: we have some other mixins. Can we update the name of this local mixin to something more specific, like `@mixin kRadioDisabled` or something similar?

The same comment applies to the other local mixins (e.g. `checked`)

Reply:
Sure, done